castervoice/rules/core/keyboard_rules/keyboard.py

```diff
@@ -30,7 +30,7 @@
 
 cat_spec_and_reverse = lambda s1, s2: "(" + s1 + " " + s2 + ")" + " | " + "(" + s2 + " " + s1 + ")"
 
-cat_spec_and_reverse_3 = lambda s1, s2, s3: cat_spec_and_reverse(cat_spec_and_reverse(s1, s2), s3) + " | (" + s1 + " " + s3 + " " + s2 + ")" + " | (" + s2 + " " + s3 + " " + s1 + ")" + " | (" + s3 + " " + s2 + " " + s1 + ")"
+cat_spec_and_reverse_3 = lambda s1, s2, s3: "(" + s1 + " (" + cat_spec_and_reverse(s2, s3) + ")) | (" + s2 + " (" + cat_spec_and_reverse(s1, s3) + ")) | (" + s3 + " (" + cat_spec_and_reverse(s1, s2) + ")"
```

Comment:
It might be helpful to add a comment in the code here to explain a little bit how this works.